rmgpy/rmg/main.py

```diff
@@ -1443,74 +1460,84 @@ def updateReactionThresholdAndReactFlags(self,
                     self.trimolecularThreshold = trimolecularThreshold
                 
             if skipUpdate:
                 return
             
             # Always update the react and threshold arrays
             for i in xrange(numCoreSpecies):
-                if not self.unimolecularThreshold[i] and rxnSysUnimolecularThreshold[i]:
-                    # We've shifted from not reacting to reacting
-                    self.unimolecularReact[i] = True
-                    self.unimolecularThreshold[i] = True
+                for k in xrange(len_families):
+                    if not self.unimolecularThreshold[i,k] and rxnSysUnimolecularThreshold[i,k]:
+                        # We've shifted from not reacting to reacting
+                        self.unimolecularReact[i,k] = True
+                        self.unimolecularThreshold[i,k] = True
 
             for i in xrange(numCoreSpecies):
                 for j in xrange(i, numCoreSpecies):
-                    if not self.bimolecularThreshold[i,j] and rxnSysBimolecularThreshold[i,j]:
-                        # We've shifted from not reacting to reacting
-                        self.bimolecularReact[i,j] = True
-                        self.bimolecularThreshold[i,j] = True
+                    for k in xrange(len_families):
+                        if not self.bimolecularThreshold[i,j,k] and rxnSysBimolecularThreshold[i,j,k]:
+                            # We've shifted from not reacting to reacting
+                            self.bimolecularReact[i,j,k] = True
+                            self.bimolecularThreshold[i,j,k] = True
 
             if self.trimolecular:
                 for i in xrange(numCoreSpecies):
                     for j in xrange(i, numCoreSpecies):
                         for k in xrange(j, numCoreSpecies):
-                            if not self.trimolecularThreshold[i,j,k] and rxnSysTrimolecularThreshold[i,j,k]:
-                                # We've shifted from not reacting to reacting
-                                self.trimolecularReact[i,j,k] = True
-                                self.trimolecularThreshold[i,j,k] = True
+                            for l in xrange(len_families):
+                                if not self.trimolecularThreshold[i,j,k,l] and rxnSysTrimolecularThreshold[i,j,k,l]:
+                                    # We've shifted from not reacting to reacting
+                                    self.trimolecularReact[i,j,k,l] = True
+                                    self.trimolecularThreshold[i,j,k,l] = True
         else:
             # We are not filtering reactions
             if new_core_species:
                 # React all the new core species unimolecularly
                 for i in xrange(prevNumCoreSpecies, numCoreSpecies):
-                    self.unimolecularReact[i] = True
+                    for k in xrange(len_families):
+                        self.unimolecularReact[i,k] = True
```

Comment:
Thanks! One thing that I need to check into is how we handle the fact that for bimolecular and trimolecular arrays that tensors are symmetric for a given family. Therefore, is `self.bimolecularReact[:, prevNumCoreSpecies:, :] = True` good enough or do we need to add an additional line of `self.bimolecularReact[prevNumCoreSpecies:, :, :] = True`?

Reply:
Just to document, we learned that these tensors are upper triangle matrices for each family slice in bimolecular, and that RMG does not use the bottom triangle to prevent reacting pairs twice.